vulcanai2/models/basenetwork.py

```diff
@@ -453,11 +454,16 @@ def forward_pass(self, input_data, convert_to_class=False):
         Returns: Numpy matrix with the output probabilities
                  with each class unless otherwise specified.
         """
-        output = self.cpu()(torch.Tensor(input_data)).data.numpy()
-        if convert_to_class:
-            return self.metrics.get_class(output)
-        else:
-            return output
+        self.eval()
+        if not isinstance(input_data, torch.Tensor):
```

Comment:
input data should be the same as fit, aka take a data_loader and perform necessary operations to get the input_data. Also add a parameter for number of samples to evaluate on.